From 184efc149a20a62a50dc7c0335befa97fd456512 Mon Sep 17 00:00:00 2001
From: Ting Chou <ting.chou@sifive.com>
Date: Wed, 7 Sep 2022 08:21:27 +0800
Subject: [PATCH] [riscv] Fix
 cctest/test-assembler-riscv*/RISCV_UTEST_FLOAT_WIDENING_vfwadd_vf.

Storing with E64 when SEW=32 has EMUL=2, which copies |n| 64 bit wide
data to the result double array already. Besides, accessing v1 when
EMUL=2 is reserved.

R=yahan@iscas.ac.cn

Change-Id: I0870d53c36b642529cab753409f52016d79219b8
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3878442
Commit-Queue: Yahan Lu <yahan@iscas.ac.cn>
Reviewed-by: Yahan Lu <yahan@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#83110}
---
 test/cctest/test-assembler-riscv32.cc | 2 --
 test/cctest/test-assembler-riscv64.cc | 2 --
 2 files changed, 4 deletions(-)

diff --git a/test/cctest/test-assembler-riscv32.cc b/test/cctest/test-assembler-riscv32.cc
index ac1f603e385..29f157055f8 100644
--- a/test/cctest/test-assembler-riscv32.cc
+++ b/test/cctest/test-assembler-riscv32.cc
@@ -2068,9 +2068,7 @@ UTEST_RVV_VF_VV_FORM_WITH_OP(vfdiv_vv, /)
       }                                                                        \
       __ instr_name(v0, v2, fa1);                                              \
       __ li(t1, Operand(int64_t(result)));                                     \
-      __ li(t2, Operand(int64_t(&result[n / 2])));                             \
       __ vs(v0, t1, 0, VSew::E64);                                             \
-      __ vs(v1, t2, 0, VSew::E64);                                             \
     };                                                                         \
     for (float rs1_fval : compiler::ValueHelper::GetVector<float>()) {         \
       for (float rs2_fval : compiler::ValueHelper::GetVector<float>()) {       \
diff --git a/test/cctest/test-assembler-riscv64.cc b/test/cctest/test-assembler-riscv64.cc
index 73e68ec9534..804ca058246 100644
--- a/test/cctest/test-assembler-riscv64.cc
+++ b/test/cctest/test-assembler-riscv64.cc
@@ -2332,9 +2332,7 @@ UTEST_RVV_VF_VV_FORM_WITH_OP(vfdiv_vv, /)
       }                                                                        \
       __ instr_name(v0, v2, fa1);                                              \
       __ li(t1, Operand(int64_t(result)));                                     \
-      __ li(t2, Operand(int64_t(&result[n / 2])));                             \
       __ vs(v0, t1, 0, VSew::E64);                                             \
-      __ vs(v1, t2, 0, VSew::E64);                                             \
     };                                                                         \
     for (float rs1_fval : compiler::ValueHelper::GetVector<float>()) {         \
       for (float rs2_fval : compiler::ValueHelper::GetVector<float>()) {       \
-- 
2.35.1

